### PR TITLE
refactor: Rework env overrides for common config in place

### PR DESCRIPTION
**Pull request by lenny-intel**
**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-compose/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) **N/A**
  <link to docs PR>

## Testing Instructions
Pull all latest images
Run the following commands and verify services bootstrap successfully and are using the appropriate MessageBus
### Secure Mode
- `make run ds-virtual ds-rest asc-sample`
  - Verify logs have : `Connected to redis Message Bus @ redis://edgex-redis:6379 with AuthMode='usernamepassword'`
- `make clean`
- `make run ds-virtual ds-rest asc-sample mqtt-bus`
  - Verify logs have : `Connected to mqtt Message Bus @ tcp://edgex-mqtt-broker:1883 with AuthMode='usernamepassword'`
- `make clean`
### Non-Secure Mode
- `make run ds-virtual ds-rest asc-sample no-secty`
  - Verify logs have : `Connected to redis Message Bus @ redis://edgex-redis:6379 with AuthMode='usernamepassword'`
- `make clean`
- `make run ds-virtual ds-rest asc-sample mqtt-bus no-secty`
  - Verify logs have : `Connected to mqtt Message Bus @ tcp://edgex-mqtt-broker:1883 with AuthMode='none'`
- `make clean`

### CODE REVIEW

**Comment by bnevis-i** [comment]
If I wanted to compare the generated docker-compose files for this vs previous version, what differences should I expect to see?

**Comment by lenny-intel** [comment]
> If I wanted to compare the generated docker-compose files for this vs previous version, what differences should I expect to see?

The diffs for the generated are in this PR. You will see many env overrides removed since they are no only needed on Common Config

File: compose-builder/add-taf-app-services.yml [hunk 1]
```diff
@@ -22,9 +22,6 @@ services:
       - 59705:59705/tcp
     container_name: app-functional-tests
     hostname: app-functional-tests
-    env_file:
-      - common.env
-      - as-common.env
```
Comment (cherrycl) [inline compose-builder/add-taf-app-services.yml:27]: Lost env_file `common-non-security.env`?

Comment (lenny-intel) [reply compose-builder/add-taf-app-services.yml]: Fix here and below. THX!

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**